contentacms/contenta_jsonapi
# Grant "access user profiles" permission to all users by default in recipes_magazin_contenta

Task: #98 

* [] Ready for merge

We want to be able to access author information as an anonymous and/or authenticated user.
This means that installing **recipes_magazin_contenta** module (which can be activated during the installation, if _Install the demo content_ checkbox is checked) will grant **View user information** ( "access user profiles") access to any user. 

### Notes
Review of the code is needed since task comment says:

> This should go into the recipes_magazin_contenta module so it can be uninstalled when the default content is yanked.

The approach followed here will grant permissions but is doing nothing when uninstalled, so I don't know if I am following the right approach, help here would be great.

### Test Instructions

1. Checkout fb-recipies-magazin-contenta branch.
2. Install profile checking _Install the demo content_ option (by default).
3. After install is complete, as an anonymous user navigate to a recipient content
4. Check that you can see the link to the user and can open that link
5. Repeat operation with an authenticated user

### QA

1. Is it the right way to implement it?
2. If this is the right way, should we aware the user that installing this module will perform this grant permissions action?

... Screenshots/Screencasts-optional
![contenta-si-01-install-demo-content](https://user-images.githubusercontent.com/1562701/35778476-f5370e5e-09be-11e8-9103-57287fdd1844.png)
![contenta-si-02-check-user-permissions-after-install](https://user-images.githubusercontent.com/1562701/35778477-f559c9ee-09be-11e8-9141-7c5cec13247e.png)
![contenta-si-03-anon-view-content-user-link](https://user-images.githubusercontent.com/1562701/35778478-f57ce4f6-09be-11e8-809c-c07e240e1f82.png)
![contenta-si-04-anon-view-user-account](https://user-images.githubusercontent.com/1562701/35778480-f59e3426-09be-11e8-8cc6-fc1e4fd84ef4.png)
